lib/msal-common/src/auth/MsalAccount.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientInfo } from "./ClientInfo";
+import { IdToken } from "./IdToken";
+import { StringUtils } from "../utils/StringUtils";
+import { StringDict } from "../utils/MsalTypes";
+import { ICrypto } from "../utils/crypto/ICrypto";
+import { IdTokenClaims } from "./IdTokenClaims";
+
+/**
+ * accountIdentifier       combination of idToken.uid and idToken.utid
+ * homeAccountIdentifier   combination of clientInfo.uid and clientInfo.utid
+ * userName                idToken.preferred_username
+ * name                    idToken.name
+ * idToken                 idToken
+ * sid                     idToken.sid - session identifier
+ * environment             idtoken.issuer (the authority that issues the token)
+ */
+export class MsalAccount {
+
+    accountIdentifier: string;
+    homeAccountIdentifier: string;
+    userName: string;
+    name: string;
+    idToken: string;
+    idTokenClaims: StringDict;
+    sid: string;
+    environment: string;
+
+    /**
+     * Creates an Account Object
+     * @praram accountIdentifier
+     * @param homeAccountIdentifier
+     * @param userName
+     * @param name
+     * @param idToken
+     * @param sid
+     * @param environment
+     */
+    constructor(accountIdentifier: string, homeAccountIdentifier: string, idTokenClaims: IdTokenClaims, rawIdToken: string) {
+        this.accountIdentifier = accountIdentifier;
+        this.homeAccountIdentifier = homeAccountIdentifier;
+        this.userName = idTokenClaims.preferred_username;
+        this.name = idTokenClaims.name;
+        // will be deprecated soon
```

Comment:
Can we remove the `idToken` here now instead of later?

Reply:
Will do some further research.

Reply:
I am going have a separate PR for Auth Code which will implement Tenant Profiles